.github/workflows/main.yml

```diff
@@ -0,0 +1,411 @@
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Development Image
+        uses: docker/build-push-action@v3
+        with:
+          tags: ${{ env.DOCKER_REGISTRY_PATH }}/${{ env.PROJECT_NAME }}-development:${{ needs.prepare-environment.outputs.project_version }}-debian
+          context: ./docker/development
+          build-args: |
+            BASE_IMAGE_VERSION=0.2.1
+            BASE_IMAGE_SYSTEM=debian
+            VERSION=${{ needs.prepare-environment.outputs.project_version }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+
+  build-image-development-fedora:
+    name: Build Development Image (Fedora)
+    runs-on: ubuntu-latest
+    needs:
+      - prepare-environment
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build Development Image
+        uses: docker/build-push-action@v3
+        with:
+          tags: ${{ env.DOCKER_REGISTRY_PATH }}/${{ env.PROJECT_NAME }}-development:${{ needs.prepare-environment.outputs.project_version }}-fedora
+          context: ./docker/development
+          build-args: |
+            BASE_IMAGE_VERSION=0.2.1
+            BASE_IMAGE_SYSTEM=fedora
+            VERSION=${{ needs.prepare-environment.outputs.project_version }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+
+  test-unit-cpp:
+    name: Run C++ Unit Tests
+    runs-on: ubuntu-latest
+    needs:
+      - prepare-environment
+      - build-image-development-debian
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Pull Development Image
+        run: docker pull ${{ env.DOCKER_REGISTRY_PATH }}/${{ env.PROJECT_NAME }}-development:${{ needs.prepare-environment.outputs.project_version }}-debian
+      - name: Run Unit Tests C++
+        run: make test-coverage-cpp-debian
+      - name: Upload Coverage Results
+        uses: codecov/codecov-action@v3
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          directory: ./coverage/
+          fail_ci_if_error: true
```

Comment:
According to the action [documentation](https://github.com/codecov/codecov-action):

`fail_ci_if_error`: Specify if CI pipeline should fail when Codecov runs into errors during upload.

So it is only to fail the job if the coverage uploads fails, not because of a coverage regression.